src/DataStructures/Tensor/Expressions/DataTypeSupport.hpp

```diff
@@ -98,9 +142,12 @@ struct upcast_if_derived_vector_type<T, true> {
 template <typename LhsDataType, typename RhsDataType>
 struct lhs_datatype_is_assignable_to_rhs_datatype_impl;
 
-/// No template specialization was matched, so it's not a known pairing
+/// If the LHS type is the complex-valued partner to the RHS type, we can assign
+/// the LHS type to the RHS type
 template <typename LhsDataType, typename RhsDataType>
-struct lhs_datatype_is_assignable_to_rhs_datatype_impl : std::false_type {};
+struct lhs_datatype_is_assignable_to_rhs_datatype_impl
+    : std::bool_constant<(
```

Comment:
once these things all inherit from `std::BLAH` you can remove the `bool_constant` stuff here too. That's why the inheritance approach is preferred, it cascades to all derived type traits